patches/minecraft/net/minecraft/entity/projectile/FishingBobberEntity.java.patch

```diff
@@ -1,5 +1,14 @@
 --- a/net/minecraft/entity/projectile/FishingBobberEntity.java
 +++ b/net/minecraft/entity/projectile/FishingBobberEntity.java
+@@ -245,7 +_,7 @@
+ 
+    private void func_190624_r() {
+       RayTraceResult raytraceresult = ProjectileHelper.func_234618_a_(this, this::func_230298_a_);
+-      this.func_70227_a(raytraceresult);
++      if (!net.minecraftforge.event.ForgeEventFactory.onProjectileImpact(this, raytraceresult)) this.func_70227_a(raytraceresult);
```

Comment:
All other projectiles only fire this event if it isn't a `MISS`. This fires the event even if it is a miss.

Reply:
I think I left that out originally because most other projectiles fire the event from inside their onHit, and I don't think I realized the non-miss check was a forge insert. I was worried about accidentally messing with/cancelling things too early or accidentally.
Fixed now though.

Reply:
Well, you are right, `func_70227_a` should still be called on `MISS`, which your patch now changes. You need an `||`, not `&&`. You want to call `func_70227_a` if it's a `MISS` _or_ the forge event was not cancelled:
`if (type == MISS || !onProjectileImpact) func_70227_a(...)`. Due to short-circuiting of `||` this will only fire the event for non-miss events still.

Reply:
Oh, that's a good use of the short-circuit, thanks! Patched now.